### ISSUE TITLE
Item display furniture hitbox issues

### ISSUE THREAD

**Issue by LuluxClient** [PluginBugs/Issues-ItemsAdder#3874]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

xdLulux

### What happened?

Since updating to the latest version of minecraft and itemsadder 4.0+. After updating the furnitures to `item_display` I've discovered lots of issues:
- Furnitures requires multiple hits to be broken
- You cannot one shot furnitures in gamemode creative and it's super hard to build with it
- The barrier blocks hitboxes are wrong, if you have a 1x1x1 hitbox furniture, if you place the furniture on a wall and sometimes on ground you will have the barrier block be 1 block higher. A patch I found is add `height_offset: -0.5` but the hitbox (the real one without the barrier block) is shifted randomly

### Steps to reproduce the issue

1. create a furniture with a 1x1x1 hitbox
2. place it on a wall
3. See a barrier block being placed on the wrong position
4. change the hight_offset to -0.5
5. issue fixed but the break hitbox is shifted sometimes
6. to break a furniture it's often super hard to break it and even harder for builders when they are in gamemode creative

### Server version

This server is running UniverseSpigot version 1.21-DEV-ver/1.21@64094d4 (2024-08-08T06:45:54Z) (Implementing API version 1.21-R0.1-SNAPSHOT)
Universe-1.21-DEV-64094d4 (MC: 1.21)

### ItemsAdder Version

4.0.2-alpha-test-3

### ProtocolLib Version

5.3.0-SNAPSHOT-726

### LoneLibs Version

1.0.56

### Full server log

private

### Error (optional)

```shell
none
```

### Problematic items yml configuration file (optional)

```shell
wisteria_yellow_vines:
    display_name: Wisteria Yellow Vines
    resource:
      material: MAGENTA_DYE
      model_id: 683
      generate: false
      model_path: epicworld/wisteria_yellow_vines
    behaviours:
      furniture:
        entity: item_display
        gravity: false
        fixed_rotation: false
        small_hitbox: false
        solid: true
        hitbox:
          length: 1
          width: 1
          height: 1
          length_offset: 0
          width_offset: 0    
          height_offset: -0.5         
        placeable_on:
          floor: false
          ceiling: true
          walls: false
        sound:
          place:
            name: block.wood.place
          break:
            name: block.wood.break
        display_transformation:
          adjust_legacy_model: item_frame
```

### Screenshots/Videos (you can drag and drop files or paste links)

Before the height offset change: https://gyazo.com/1283b5357ecc0912f390cf935c2de92a

After the height offset change: https://gyazo.com/0cf0919f42d63f4eacbe603bd906ce7d

**Comment by SrMonsterYT** [comment]
same problem as you

**Comment by LoneDev6** [comment]
> Furnitures requires multiple hits to be broken
this is normal, it's to give players feedback, like vanilla armorstands

> You cannot one shot furnitures in gamemode creative and it's super hard to build with it 
will be fixed, created a new issue for that #4122

> The barrier blocks hitboxes are wrong, if you have a 1x1x1 hitbox furniture, if you place the furniture on a wall and sometimes on ground you will have the barrier block be 1 block higher. A patch I found is add height_offset: -0.5 but the hitbox (the real one without the barrier block) is shifted randomly

Should be fixed in the latest beta 4.0.4, please join the Discord server and verify your purchase, then access the beta channel.
